crates/core/src/db/relational_db.rs

```diff
@@ -61,9 +67,12 @@ pub struct RelationalDB {
     /// is `Some`, `None` otherwise.
     disk_size_fn: Option<DiskSizeFn>,
 
-    // Release file lock last when dropping.
-    _lock: Arc<File>,
     config: Arc<RwLock<DatabaseConfig>>,
+
+    // DO NOT ADD FIELDS AFTER THIS.
+    // By default, fields are dropped in declaration order.
+    // We want to release the file lock last.
+    _lock: Arc<File>,
```

Comment:
This is pretty nasty, tbh. I know it's not your doing, and I don't consider this a blocker, but you might consider writing a manual `impl Drop` rather than relying on the automatic one in order to be more explicit about this constraint.

Reply:
Actually, dropping some field _first_ in an `impl Drop` I know how to do, but not how to drop it _last_. Would it not require to wrap all the fields in `ManuallyDrop`?

Reply:
Oh, right, I forgot that even when you manually `impl Drop` you still get the auto-generated drop method as an epilogue. Disregard.